src/api/dist_communicator.i

```diff
@@ -44,7 +44,7 @@ public:
   int totalMPIRanksInGlobal;
   int MPIRankInLocal;
   Communicator(int limit);
-  Communicator(int gpu_num, int gpu_per_node, const NcclIdHolder &holder, int limit);
+  Communicator(int gpu_num, int num_gpus, const NcclIdHolder &holder, int limit);
```

Comment:
I think we can unify the terms, e.g., using <worker_id, num_workers> or <rank, local_rank and world_size>
the first set is for distributed training; the second set is for communication. maybe the second set is better.

Reply:
how about I change all the code including python and C++ to <rank, local_rank, world_size> ?

Reply:
ok.